pkg/filestore/writer.go

```diff
@@ -74,6 +75,28 @@ func (w *Writer) WriteKeypair(meta metadata.Metadata, key crypto.PrivateKey, cha
 
 	keyPEM := pem.EncodeToMemory(pemBlock)
 
+	files := map[string][]byte{
+		attrs[csiapi.KeyFileKey]:  keyPEM,
+		attrs[csiapi.CertFileKey]: chain,
+		attrs[csiapi.CAFileKey]:   ca,
+	}
+
+	if attrs[csiapi.KeystoreFile] != "" {
+		switch keyStoreType := attrs[csiapi.KeystoreType]; keyStoreType {
+		case "PKCS12":
+			pfx, err := pkcs12.Create(key, chain, ca)
+			if err != nil {
+				return fmt.Errorf("pkcs12.Create: %v", err)
+			}
+
+			files[attrs[csiapi.KeystoreFile]] = pfx
+			meta.VolumeContext[csiapi.KeystoreFile] = attrs[csiapi.KeystoreFile]
+			meta.VolumeContext[csiapi.KeystoreType] = keyStoreType
```

Comment:
How come we need to write to the VolumeContext here? I don't see other occurrences of writing to the VolumeContext in this function so would be good to include a comment if it is needed (and are there any other places we _do_ already write to the VolumeContext and would it be more suitable to do it there instead?)

Reply:
Good catch - I didn't realise the that we will always write non-default attributes as metadata and these are superfluous. Fixed!